File: app/src/main/java/com/starostinvlad/fan/SearchedDao.java
package com.starostinvlad.fan;

import com.starostinvlad.fan.GsonModels.News;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import io.reactivex.Maybe;
import io.reactivex.Observable;

@Dao
public interface SearchedDao {

    @Query("SELECT * FROM news")
    Observable<List<News>> getAll();

    @Query("SELECT * FROM news WHERE id = :id")
    News getById(long id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Maybe<List<Long>> insert(List<News> searched);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Maybe<Long> insert(News searched);

    @Update
    void update(News searched);

    @Update
    void update(List<News> searched);

    @Delete
    void delete(News searched);

}

File: app/src/main/java/com/starostinvlad/fan/SearchedDao.kt
package com.starostinvlad.fan

import androidx.room.*
import com.starostinvlad.fan.GsonModels.News
import io.reactivex.Maybe
import io.reactivex.Observable

@Dao
interface SearchedDao {
    @get:Query("SELECT * FROM news")
    val all: Maybe<List<News>>

    @Query("SELECT * FROM news WHERE id = :id")
    fun getById(id: Long): News

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(searched: List<News>): Maybe<List<Long>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(searched: News): Maybe<Long>

    @Update
    fun update(searched: News)

    @Update
    fun update(searched: List<News>)

    @Delete
    fun delete(searched: News)
}